### ISSUE TITLE
Why choose to train each module individually

### ISSUE THREAD

**Issue by centiLinda** [ApeironY/Modular-Trajectory-Prediction-Toolkit#3]
Hi, thanks for sharing your work!

If I understand your code correctly, each module of your network is trained and saved sequentially (past_encoder, future_encoder, classifier, etc.), rather than jointly trained. May I know why you choose to design it this way?

Usually all modules in a network are jointly trained to serve the overall loss function (in our case the error between ground truth and predicted trajectory). So the classifier, for example, will learn to classify a trajectory such that it leads to a more accurate prediction from the later decoder. In your design, the classifier learns to reproduce the results from k-means clustering. But this might not necessarily be useful for trajectory prediction.

Take image classification as an example, the encoder and classifier need to be jointly trained. If you choose to train them sequentially, let's say you use a convo-deconvo network to train a cnn encoder, it might be good at regenerating the image, but it cannot encode the image into useful tensors that can be used in the later classifier.

I hope I have explained it clearly. Understand that this work is from quite a while ago, but I would love to hear your opinion.
